pragma-node/src/handlers/entries/get_onchain/mod.rs

```diff
@@ -33,20 +33,33 @@ pub async fn get_onchain(
     State(state): State<AppState>,
     PathExtractor(pair): PathExtractor<(String, String)>,
     Query(params): Query<GetOnchainParams>,
-) -> Result<Json<GetOnchainResponse>, EntryError> {
+) -> Result<Json<Vec<GetOnchainResponse>>, EntryError> {
     tracing::info!("Received get onchain entry request for pair {:?}", pair);
     let is_routing = params.routing.unwrap_or(false);
 
     let pair_id: String = currency_pair_to_pair_id(&pair.0, &pair.1);
     let now = chrono::Utc::now().timestamp() as u64;
     let aggregation_mode = params.aggregation.unwrap_or_default();
-    let timestamp = if let Some(timestamp) = params.timestamp {
-        if timestamp > now {
-            return Err(EntryError::InvalidTimestamp);
+    let timestamp = match params.timestamp {
```

Comment:
This should definitely have its own function (makes it also usable for other endpoints if needed) and we call it to convert `params.timestamp` -> `Timestamp`

Reply:
done